hystrix-contrib/hystrix-javanica/hystrix-javanica-core/src/main/java/com/netflix/hystrix/contrib/javanica/command/closure/BaseClosureFactory.java

```diff
@@ -60,27 +33,29 @@ public Closure createClosure(MetaHolder metaHolder, Method method, Object o, Obj
      * @return new {@link Closure} instance
      * @throws Exception
      */
-    Closure createClosure(String rootMethodName, final Object closureObj) throws Exception {
+   protected Closure createClosure(String rootMethodName, final Object closureObj) throws Exception {
         if (!isClosureCommand(closureObj)) {
             throw new RuntimeException(format(ERROR_TYPE_MESSAGE, rootMethodName,
                     getClosureCommandType().getName()).getMessage());
         }
         Method closureMethod = closureObj.getClass().getMethod(INVOKE_METHOD);
         return new Closure(closureMethod, closureObj);
     }
+    
+    protected abstract Object execute(T metaHolder, Method method, Object o, Object... args);
```

Comment:
I think it makes sense to remove `ClosureFactory` and its implementations. It is used only by `MethodExecutionAction` and `AopAllianceMethodExecutionAction`. The `execute()` part in `BaseClosureFactory` is already present in these classes for other purposes. So it makes sense to create a common class `AbstractMethodExecutionAction`, keep the closure creation and other common stuffs here and let   `MethodExecutionAction` and `AopAllianceMethodExecutionAction` subclass this class. This is something you already suggested in one of the review comments but I failed to think in this direction.